src/sets.jl

```diff
@@ -669,18 +669,69 @@ end
 
 Base.:(==)(set1::IndicatorSet{A, S}, set2::IndicatorSet{A, S}) where {A, S} = set1.set == set2.set
 
+"""
+    Complements(dimension::Int)
+
+The set corresponding to a mixed complementarity constraint.
+
+Complementarity constraints should be specified with an
+[`AbstractVectorFunction`](@ref)-in-`Complements(dimension)` constraint.
+
+The dimension of the vector-valued function `F` must be `2 * dimension`. This
+defines a complementarity constraint between the scalar function `F[i]` and the
+variable in `F[i + dimension]`. Thus, `F[i + dimension]` must be interpretable
+as a single variable `x_i` (e.g., `1.0 * x + 0.0`).
+
+If the variable `x_i` is constrained in `Interval(lb_i, ub_i)`, then
+mathematically, the mixed complementarity problem is to find a solution such
+that at least one of the following holds:
+
+  1. `F_i(x) == 0` if `lb_i < x_i < ub_i`
+  2. `F_i(x) >= 0` if `lb_i == x_i`
+  3. `F_i(x) <= 0` if `x_i == ub_i`
+
+Classically, the bounding set for `x_i` is `Interval(0, Inf)`, which recovers:
+`0 <= F_i(x) ⟂ x_i >= 0`, where the `⟂` operator implies `F_i(x) * x_i = 0`.
+
+### Examples
+
+The problem:
+
+    x -in- Interval(-1, 1)
+    [2 * x - 3, x] -in- Complements(1)
+
+defines the mixed complementarity problem where at least one of the following
+holds:
+
+  1. `2 * x - 3 == 0` if `-1 < x < 1`
+  2. `2 * x - 3 >= 0` if `x == -1`
+  3. `2 * x - 3 <= 0` if `x == 1`
+
```

Comment:
Here also, all the statements must hold. Note that this is more apparent in the non-monotone case.

Suppose that the problem is `[-4 * x - 3, x] -in- Complements(1)`. Then there are 3 solutions:
1. x == -3/4; F(x) = 0
2. x == -1; F(x) = 1
3. x == 1; F(x) = -7

So all statements must hold to be equivalent to the MCP. Maybe it is even better to have that example with all the solution in the documentation?